include/boost/geometry/formulas/area_formulas.hpp

```diff
@@ -472,43 +487,46 @@ class area_formulas
 
         CT excess;
 
-        auto const half_pi = math::pi<CT>() / 2;
-        bool meridian = lon2r - lon1r == CT(0)
-            || lat1r == half_pi || lat1r == -half_pi
-            || lat2r == half_pi || lat2r == -half_pi;
+        CT lon12r = lon2r - lon1r;
+        math::normalize_longitude<radian, CT>(lon12r);
 
-        if (!meridian && (i_res.distance)
-            < mean_radius<CT>(spheroid_const.m_spheroid) / CT(638))  // short segment
+        if (lon12r == pi || lon12r == -pi)
```

Comment:
Is a double comparison `==` OK in this case?
Probably, but maybe you can add a comment

Reply:
OK I added a comment